### ISSUE TITLE
bug: 10.0.2 regression

### ISSUE THREAD

**Issue by shrugs** [GetStream/stream-chat-react#1757]
**Describe the bug**

the html generated by stream-chat-react is producing a div with the class names `type className selector props toString messaging light type className selector props toString` which does not have styling and breaks layout. doesn't occur with 10.0.1

<img width="543" alt="image" src="https://user-images.githubusercontent.com/1535001/190826552-5986124c-5d23-429e-b630-f3ba7fa1b90f.png">

i believe it's this div in `Channel`: https://github.com/GetStream/stream-chat-react/blob/03fbe2b5ab1f71d82880586bada6ed2cba156db4/src/components/Channel/Channel.tsx#L932

**Comment by shrugs** [comment]
`c-bjFCTN` is `.chatContainer` and `c-kZkohQ` is `.messageList`

**Comment by arnautov-anton** [comment]
Hey, @shrugs, thank you for submitting! Are you using @emotion-js or something similar by any chance? It looks like the `clsx` function is receiving an object and instead of calling `toString` on it, it generates the className string (from that object properties) based on whether property values of that object are truthy. This one is definitely an unexpected one but it would be great if you could point me in the right direction so I could reproduce it, thank you.

**Comment by shrugs** [comment]
i am using stitches, very similar to emotion/styled, and passing some custom class names that it generates, though they should be compiled to a string by the time stream-chat-react sees them. any changes in 10.0.1->10.0.2 that might be related? can cross reference that with how my app renders the components

**Comment by arnautov-anton** [comment]
Just managed to reproduce it with this piece of code using `@stitches/react`:

<img width="547" alt="image" src="https://user-images.githubusercontent.com/43254280/190924206-eadf5199-4c99-4030-a379-b6cd4320b36d.png">

And the `channel` generated with the `css` function is truly an object which has `toString` method on it:

<img width="518" alt="image" src="https://user-images.githubusercontent.com/43254280/190924861-493ac241-816a-4b4b-a024-aafb8eee1d69.png">

The `customClasses` property expects a record of pre-specified keys with **string** type values (class names). We did [a refactor](https://github.com/GetStream/stream-chat-react/pull/1747/commits/0a79c5ccb920c02dd4e47820849b50873a710ee3) to the way the Channel component handles these strings - which now is done with the help of `clsx` which treats objects the way I described previously. I will look into making `customClasses` object stricter type-wise if that's possible but nothing from the way we treat custom class names (as strings) has changed. I'll also discuss with the team an option of adding `toString?.()` calls to custom classes to handle such edge-cases but for now I'd recommend adding `.toString()` to those objects yourself.
